### PR TITLE
Fix for the issue #71

### PR DESCRIPTION
**Pull request by wakr**
https://github.com/testmycode/tmc-core/issues/71

Removed all imports to local Validations & ValidationError and also removed them completely. They weren't in use though?

### CODE REVIEW

File: src/main/java/fi/helsinki/cs/tmc/core/domain/submission/SubmissionResult.java [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
+import com.google.gson.annotations.SerializedName;
+
 import java.util.Collections;
 import java.util.List;
 
+/**
+ * This class is missing tests at the moment.
+ * All domain.submission -classes are missing tests also and
+ * this class contains lots of uncommented code plus
+ * lots of methods that are unused atm.
```
Comment (jamo) [inline src/main/java/fi/helsinki/cs/tmc/core/domain/submission/SubmissionResult.java]: I don't feel like this should be javadoc...

File: src/main/java/fi/helsinki/cs/tmc/core/domain/submission/SubmissionResult.java [hunk 2]
```diff
@@ -104,14 +102,6 @@ public void setError(String error) {
         this.error = error;
     }
 
-    public Validations getValidations() {
-        return validations;
-    }
-
-    public void setValidations(Validations validations) {
```
Comment (jamo) [inline src/main/java/fi/helsinki/cs/tmc/core/domain/submission/SubmissionResult.java:111]: I'm wondering whether these would actually be needed; I recall that this class is instantiated by a json library and server does return validation results also when exercises are configured to run validations (e.g checkstyle)

Comment (ljleppan) [inline src/main/java/fi/helsinki/cs/tmc/core/domain/submission/SubmissionResult.java:111]: We should get @nygrenh to clarify this

Comment (jamo) [inline src/main/java/fi/helsinki/cs/tmc/core/domain/submission/SubmissionResult.java:111]: These indeed seem to be the classes from NetBeans and thus representing the submission result, which sometimes does contain submission results.

Comment (jamo) [inline src/main/java/fi/helsinki/cs/tmc/core/domain/submission/SubmissionResult.java:111]: So let's restore these validations

**Comment by wakr** [comment]
Restoring Validations-class and Validation-error class would mean that this pr does nothing. 

"We want to replace fi.helsinki.cs.tmc.core.domain.submission.ValidationError and fi.helsinki.cs.tmc.core.domain.submission.Validations with the classes provided by TMC-Langs-abstraction."

^ where does this apply now if we need to restore original validations-class?

**Comment by jamo** [comment]
We don't want to restore the Validation classes, but not remove the unused validation fields of SubmissionResult

**Comment by wakr** [comment]
If Validation classes are removed SubmissionResult-class cannot contain any reference to it. For example `private Validations validations;` would of course be illegal line because there is no more import for it (locating actually in the same package). Do you mean restoring [this](https://github.com/wakr/tmc-core/blob/0a8d07381b24ad11a9aab03d0861a8c99b1d1253/src/main/java/fi/helsinki/cs/tmc/core/domain/submission/SubmissionResult.java#L68) and [this](https://github.com/wakr/tmc-core/blob/0a8d07381b24ad11a9aab03d0861a8c99b1d1253/src/main/java/fi/helsinki/cs/tmc/core/domain/submission/SubmissionResult.java#L326)?

**Comment by jamo** [comment]
Could we maybe import those from tmc-langs-abstraction?

**Comment by wakr** [comment]
Yes. We could use tmc-langs-abstraction which provides interfaces and just import them to replace Validations-class (Validations => ValidationResult). Will do that next and see if there are any problems with that approach.

**Comment by wakr** [comment]
There already is a import for tmc-langs-abstraction. @jamox see [this](https://github.com/wakr/tmc-core/blob/validations_fix/src/main/java/fi/helsinki/cs/tmc/core/domain/submission/SubmissionResult.java#L3), [this](https://github.com/wakr/tmc-core/blob/validations_fix/src/main/java/fi/helsinki/cs/tmc/core/domain/submission/SubmissionResult.java#L78) and [this](https://github.com/wakr/tmc-core/blob/validations_fix/src/main/java/fi/helsinki/cs/tmc/core/domain/submission/SubmissionResult.java#L323).

**Comment by jamo** [comment]
I feel like the diff is not showing the whole truth..

**Comment by jamo** [comment]
I just confused the getValidations and getValidationResults - but why we had both in the first place?

**Comment by jamo** [comment]
Well that field is not in our [original code](https://github.com/testmycode/tmc-netbeans/blob/master/tmc-plugin/src/fi/helsinki/cs/tmc/data/SubmissionResult.java)
The only thing what I'm wondering is that the json we get from the server has field [validations](https://github.com/testmycode/tmc-server/blob/master/app/controllers/submissions_controller.rb#L53) and how gson sets the correct value if the name is different

**Pull request closed without merging**